src/main/java/org/jenkinsci/plugins/docker/workflow/WithContainerStep.java

```diff
@@ -186,7 +187,7 @@ private static void destroy(String container, Launcher launcher, Node node, EnvV
                         throw new IOException("failed to run ps");
                     }
                     List<String> pids = new ArrayList<String>();
-                    LINE: for (String line : baos.toString().split("\n")) {
+                    LINE: for (String line : baos.toString(Charset.defaultCharset().name()).split("\n")) {
                         for (Map.Entry<String,String> entry : modelEnvVars.entrySet()) {
```

Comment:
`Charset.defaultCharset()` is probably worse than useless, since this code runs on the master, whereas the text is coming from a Docker container in turn on a slave. Anyway it makes no difference here which 8-bit encoding you are using since we are only searching for a value known to be ASCII.